### PR TITLE
Prevented empty messages from being sent.

### PR DESCRIPTION
**Pull request by nedp**
Resolves issue #21 and minor miscelaneous issues:

* Removed extraneous variable in `Messages::sendMessage`
* Moved `Messages::receiveMessage` code into a
  $rootScope.$apply block for consistency.
* Initialised `MessageCtrl.newMessage` to ''.
* Added code an early return to `MessageCtrl#sendMessage`
  when `newMessage` with whitespace trimmed is empty.
* Added tests for this change.

### CODE REVIEW

**Comment by rongduan-zhu** [comment]
Tests are failing. 
```bash
Chromium 41.0.2272 (Ubuntu 0.0.0) Messages receiveMessage registers the messages in order FAILED
	Expected 0 to be 1.
	    at /home/rongduan-zhu/Documents/projects/neroma-chat/statics/js/tests/message.spec.js:104:38
	    at Array.forEach (native)
	    at Object.<anonymous> (/home/rongduan-zhu/Documents/projects/neroma-chat/statics/js/tests/message.spec.js:98:17)
	TypeError: Cannot read property 'message' of undefined
	    at /home/rongduan-zhu/Documents/projects/neroma-chat/statics/js/tests/message.spec.js:105:47
	    at Array.forEach (native)
	    at Object.<anonymous> (/home/rongduan-zhu/Documents/projects/neroma-chat/statics/js/tests/message.spec.js:98:17)
```
After some investigation, it seems that when we wrap the block of code into ```$rootScope.$apply()```, the wrapped code is called asynchronously, so when ```expect(Messages.data.length).toBe(expectedLength);``` is called, since function argument for ```$apply``` is called asynchronously, the expected length is different, thus causing the tests to fail. So I think you its fine in this case to put it outside of the ```$apply()``` method, or the alternative it the change the test to account for this.

**Comment by nedp** [comment]
My mistake, I didn't check the tests after swapping the `$apply` style.
In this case I don't think the enclosed code can throw (bar a logic error) but in general how would you go about changing the test to account for this?

**Comment by nedp** [comment]
Any outstanding issues with this?

**Comment by rongduan-zhu** [comment]
:+1:

File: statics/js/tests/message.spec.js [hunk 1]
```diff
@@ -2,12 +2,13 @@ describe('MessageCtrl', function() {
...
   var Messages;
   var User;
+  var MessageCtrl;
 
   // use chat module and mock Messages services
   beforeEach(module('chat', function($provide) {
     $provide.value('Messages', {
         data: [],
-        sendMessage: jasmine.createSpy('send').and.callFake(function() {}),
+        sendMessage: jasmine.createSpy('sendMessage').and.callFake(function() {}),
```
Comment (Druddigon) [inline statics/js/tests/message.spec.js]: Do we need `and.callFake` here?

Comment (nedp) [inline statics/js/tests/message.spec.js]: Apparently not; was following the convention already in place.

Should we remove empty fake implementations in general?

**Pull request closed without merging**